client/files.go

```diff
@@ -575,28 +575,30 @@ func (client *Client) Pull(opts *PullOptions) error {
 	if mediaType != "multipart/form-data" {
 		// If it's not multipart, it's likely an API error, so try parsing it as such.
 		var serverResp response
-		err := decodeInto(resp.Body, &serverResp)
-		if err != nil {
+		if err := decodeInto(resp.Body, &serverResp); err != nil {
 			return fmt.Errorf("expected a multipart response, got %q", mediaType)
 		}
-		err = serverResp.err()
-		if err != nil {
+		if err := serverResp.err(); err != nil {
 			return err
 		}
 		return fmt.Errorf("expected a multipart response, got %q", mediaType)
 	}
 
 	mr := multipart.NewReader(resp.Body, params["boundary"])
-	filesPart, err := mr.NextPart()
+	firstPart, err := mr.NextPart()
 	if err != nil {
 		return fmt.Errorf("cannot decode multipart payload: %w", err)
 	}
-	defer filesPart.Close()
+	defer firstPart.Close()
 
-	if filesPart.FormName() != "files" {
-		return fmt.Errorf(`expected first field name to be "files", got %q`, filesPart.FormName())
+	// Check if we got "response" first (error case) or "files" (success case)
+	if firstPart.FormName() == "response" {
+		// Decode and process the response to extract the actual error
+		return processResponsePart(firstPart)
```

Comment:
This seems reasonable. But after this `if` statement I think we should keep the `if firstPart.FormName() != "files"` block in case the field name is something other than `files` or `response`.